docs/platform/visual/samples/bar-d3-sandbox/step3-view-creation.md

```diff
@@ -11,37 +11,34 @@ layout: default
 Create a file named `view-d3.js` and place the following code in it:
 
 ```js
-define([
-  "module",
-  "pentaho/visual/base/view",
-  "./model",
-  "d3"
-], function(module, baseViewFactory, barModelFactory, d3) {
+define(["module", "d3"], function(module, d3) {
 
   "use strict";
 
-  return function(context) {
-
-    var BaseView = context.get(baseViewFactory);
-
-    var BarView = BaseView.extend({
-      $type: {
-        id: module.id,
-        props: [
-          {
-            name: "model",
-            type: barModelFactory
-          }
-        ]
-      },
+  return [
+    "pentaho/visual/base/view",
```

Comment:
Are these AMD module IDs? If yes, why not include "d3" in here? If not, and they are exclusively type IDs (or expressions), then the syntax should become clearly different (like ```"pentaho.visual.samples.bar.Model"```).

Very confusing, altogether.

Reply:
Any way to inject a reference to the context itself?

Are type aliases supported? Could I just inject ```"="```?

Reply:
Also, forgot to ask: why don't type factories receive the config and do whatever they want with it, like instances?

Reply:
1. Yes, they're AMD module IDs, which, at the moment, are the identifiers of Type API types and instances. Only Type API type dependencies are included here. Other regular AMD dependencies are left where they were.

Reply:
For the record. Like we talked, in the future we should revisit the way that we are declaring dependencies in multiple ways and check if we can reduce them (to whichever is better).  

I personally would also like to revisit the way (syntax) that we ask for multiple instances of a given type.